File: src/com/markupartist/sthlmtraveling/provider/deviation/Deviation.java
package com.markupartist.sthlmtraveling.provider.deviation;

import android.os.Parcel;
import android.os.Parcelable;
import android.text.format.Time;

public class Deviation implements Parcelable {
    private Long mReference;
    private int mMessageVersion;
    private String mLink;
    private String mMobileLink;
    private Time mCreated;
    private boolean mIsMainNews;
    private int mSortOrder;
    private String mHeader;
    private String mDetails;
    private String mScope;
    private String mScopeElements;

    public Deviation(Parcel parcel) {
        mReference = parcel.readLong();
        mMessageVersion = parcel.readInt();
        mLink = parcel.readString();
        mMobileLink = parcel.readString();
        mCreated = new Time();
        mCreated.parse3339(parcel.readString());
        mIsMainNews = (Boolean) parcel.readValue(null);
        mSortOrder = parcel.readInt();
        mHeader = parcel.readString();
        mDetails = parcel.readString();
        mScope = parcel.readString();
        mScopeElements = parcel.readString();
    }

    public Deviation() {
    }

    public void setReference(Long reference) {
        this.mReference = reference;
    }
    public Long getReference() {
        return mReference;
    }
    public void setMessageVersion(int messageVersion) {
        this.mMessageVersion = messageVersion;
    }
    public int getMessageVersion() {
        return mMessageVersion;
    }
    public void setMobileLink(String mobileLink) {
        this.mMobileLink = mobileLink;
    }
    public String getMobileLink() {
        return mMobileLink;
    }
    public void setCreated(Time created) {
        this.mCreated = created;
    }
    public Time getCreated() {
        return mCreated;
    }
    public void setMainNews(boolean isMainNews) {
        this.mIsMainNews = isMainNews;
    }
    public boolean isMainNews() {
        return mIsMainNews;
    }
    public void setSortOrder(int sortOrder) {
        this.mSortOrder = sortOrder;
    }
    public int getSortOrder() {
        return mSortOrder;
    }
    public void setLink(String mLink) {
        this.mLink = mLink;
    }
    public String getLink() {
        return mLink;
    }
    public void setHeader(String header) {
        this.mHeader = header;
    }
    public String getHeader() {
        return mHeader;
    }
    public void setDetails(String details) {
        this.mDetails = details;    
    }
    public String getDetails() {
        return mDetails;
    }
    public void setScope(String scope) {
        this.mScope = scope;
    }
    public String getScope() {
        return mScope;
    }
    public void setScopeElements(String scopeElements) {
        this.mScopeElements = scopeElements;
    }
    public String getScopeElements() {
        return mScopeElements;
    }
    
    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeLong(mReference);
        parcel.writeInt(mMessageVersion);
        parcel.writeString(mLink);
        parcel.writeString(mMobileLink);
        parcel.writeString(mCreated.format3339(false));
        parcel.writeValue(mIsMainNews);
        parcel.writeInt(mSortOrder);
        parcel.writeString(mHeader);
        parcel.writeString(mDetails);
        parcel.writeString(mScope);
        parcel.writeString(mScopeElements);
    }

    public static final Creator<Deviation> CREATOR = new Creator<Deviation>() {
        public Deviation createFromParcel(Parcel parcel) {
            return new Deviation(parcel);
        }

        public Deviation[] newArray(int size) {
            return new Deviation[size];
        }
    };
    
    /* (non-Javadoc)
     * @see java.lang.Object#toString()
     */
    @Override
    public String toString() {
        return "Deviation [mCreated=" + mCreated + ", mDetails=" + mDetails
                + ", mHeader=" + mHeader + ", mIsMainNews=" + mIsMainNews
                + ", mLink=" + mLink + ", mMessageVersion=" + mMessageVersion
                + ", mMobileLink=" + mMobileLink + ", mReference=" + mReference
                + ", mScope=" + mScope + ", mSortOrder=" + mSortOrder
                + ", sScopeElements=" + mScopeElements + "]";
    }
}

File: sthlmtraveling/src/main/java/com/markupartist/sthlmtraveling/provider/deviation/Deviation.kt
package com.markupartist.sthlmtraveling.provider.deviation

import android.os.Parcel
import android.os.Parcelable
import android.text.format.Time

class Deviation : Parcelable {
    private var mReference: Long? = null
    var messageVersion: Int = 0
    var link: String? = null
    var mobileLink: String? = null
    private var mCreated: Time? = null
    var isMainNews: Boolean = false
    var sortOrder: Int = 0
    var header: String? = null
    var details: String? = null
    var scope: String? = null
    var scopeElements: String? = null

    constructor(parcel: Parcel) {
        mReference = parcel.readLong()
        this.messageVersion = parcel.readInt()
        this.link = parcel.readString()
        this.mobileLink = parcel.readString()
        mCreated = Time()
        mCreated!!.parse3339(parcel.readString())
        this.isMainNews = (parcel.readValue(null) as kotlin.Boolean?)!!
        this.sortOrder = parcel.readInt()
        this.header = parcel.readString()
        this.details = parcel.readString()
        this.scope = parcel.readString()
        this.scopeElements = parcel.readString()
    }

    constructor()

    var reference: Long
        get() = mReference!!
        set(reference) {
            this.mReference = reference
        }
    var created: Time?
        get() = mCreated
        set(created) {
            this.mCreated = created
        }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeLong(mReference!!)
        parcel.writeInt(this.messageVersion)
        parcel.writeString(this.link)
        parcel.writeString(this.mobileLink)
        parcel.writeString(mCreated!!.format3339(false))
        parcel.writeValue(this.isMainNews)
        parcel.writeInt(this.sortOrder)
        parcel.writeString(this.header)
        parcel.writeString(this.details)
        parcel.writeString(this.scope)
        parcel.writeString(this.scopeElements)
    }

    /* (non-Javadoc)
     * @see java.lang.Object#toString()
     */
    override fun toString(): String {
        return ("Deviation [mCreated=" + mCreated + ", mDetails=" + this.details
                + ", mHeader=" + this.header + ", mIsMainNews=" + this.isMainNews
                + ", mLink=" + this.link + ", mMessageVersion=" + this.messageVersion
                + ", mMobileLink=" + this.mobileLink + ", mReference=" + mReference
                + ", mScope=" + this.scope + ", mSortOrder=" + this.sortOrder
                + ", sScopeElements=" + this.scopeElements + "]")
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Deviation?> = object : Parcelable.Creator<Deviation?> {
            override fun createFromParcel(parcel: Parcel): Deviation {
                return Deviation(parcel)
            }

            override fun newArray(size: Int): Array<Deviation?> {
                return arrayOfNulls<Deviation>(size)
            }
        }
    }
}
